src/components/spreadsheet/equipment-table.js

```diff
@@ -29,8 +35,74 @@ export const EquipmentTable = ({
     fetched,
     network,
     shouldHidePinnedHeaderRightBorder,
+    onCellClicked,
 }) => {
     const theme = useTheme();
+    const { snackError } = useSnackMessage();
+    const [clickedCellData, setClickedCellData] = React.useState({});
+    const [generatorId, setGeneratorId] = React.useState('');
+    const [
+        popupGeneratorEditRegulatingTerminalFields,
+        setPopupGeneratorEditRegulatingTerminalFields,
+    ] = React.useState(false);
+    const onRegulatingTerminalPopupClose = () => {
+        setPopupGeneratorEditRegulatingTerminalFields(false);
+    };
+    const handleSavePopupGeneratorEditRegulatingTerminalFields = (
+        voltageRegulationGenerator
+    ) => {
+        setGeneratorId(clickedCellData.data.id);
+
+        const isDistantRegulation =
+            voltageRegulationGenerator.voltageRegulationType ===
+            REGULATION_TYPES.DISTANT.id;
+
+        modifyGenerator(
+            studyUuid,
```

Comment:
**modifyGenerator** must absolutely be changed.
To check with the techleads if it should be done now or later?